rust/src/tests/chown.rs

```diff
@@ -0,0 +1,1074 @@
+
+    // Test if it follows symlinks
+
+    let link = ctx.create(FileType::Symlink(Some(file.clone()))).unwrap();
+
+    assert_ctime_changed(ctx, &link, || {
+        assert!(chown(&link, Some(user.uid), Some(group.gid)).is_ok());
+    });
+    let link_stat = metadata(&link).unwrap();
+    assert_eq!(link_stat.uid() as uid_t, user.uid.as_raw());
+    assert_eq!(link_stat.gid() as gid_t, group.gid.as_raw());
+
+    assert_ctime_changed(ctx, &link, || {
+        ctx.as_user(&user, Some(&[other_group.gid]), || {
+            assert!(chown(&link, Some(user.uid), Some(other_group.gid)).is_ok());
+        });
+    });
+    let link_stat = metadata(&link).unwrap();
+    assert_eq!(link_stat.uid() as uid_t, user.uid.as_raw());
+    assert_eq!(link_stat.gid() as gid_t, other_group.gid.as_raw());
+
+    // lchown
+    assert_ctime_changed(ctx, &file, || {
+        assert!(lchown(&file, Some(user.uid), Some(group.gid)).is_ok());
+    });
+    let file_stat = metadata(&file).unwrap();
+    assert_eq!(file_stat.uid() as uid_t, user.uid.as_raw());
+    assert_eq!(file_stat.gid() as gid_t, group.gid.as_raw());
+
+    assert_ctime_changed(ctx, &file, || {
+        ctx.as_user(&user, Some(&[other_group.gid]), || {
+            assert!(lchown(&file, Some(user.uid), Some(other_group.gid)).is_ok());
+        });
+    });
+    let file_stat = metadata(&file).unwrap();
+    assert_eq!(file_stat.uid() as uid_t, user.uid.as_raw());
+    assert_eq!(file_stat.gid() as gid_t, other_group.gid.as_raw());
+}
+
+#[cfg(any(target_os = "freebsd", target_os = "netbsd", target_os = "dragonfly"))]
+crate::test_case! {
+    /// successful chown(2) call (except uid and gid equal to -1) updates ctime.
+    update_ctime_success_symlink, serialized, root
+}
+#[cfg(any(target_os = "freebsd", target_os = "netbsd", target_os = "dragonfly"))]
+fn update_ctime_success_symlink(ctx: &mut SerializedTestContext) {
+    use super::assert_symlink_ctime_changed;
+
+    let file = ctx.create(FileType::Symlink(None)).unwrap();
+    let (user, group) = ctx.get_new_entry();
+    let other_group = ctx.get_new_group();
+
+    assert_symlink_ctime_changed(ctx, &file, || {
+        assert!(lchown(&file, Some(user.uid), Some(group.gid)).is_ok());
+    });
+    let file_stat = symlink_metadata(&file).unwrap();
+    assert_eq!(file_stat.uid() as uid_t, user.uid.as_raw());
+    assert_eq!(file_stat.gid() as gid_t, group.gid.as_raw());
+
+    assert_symlink_ctime_changed(ctx, &file, || {
+        ctx.as_user(&user, Some(&[other_group.gid]), || {
+            assert!(lchown(&file, Some(user.uid), Some(other_group.gid)).is_ok());
+        });
+    });
+    let file_stat = symlink_metadata(&file).unwrap();
+    assert_eq!(file_stat.uid() as uid_t, user.uid.as_raw());
+    assert_eq!(file_stat.gid() as gid_t, other_group.gid.as_raw());
+}
+
+// TODO: On Linux: According to POSIX: If both owner and group are -1, the times need not be updated.
+#[cfg(not(target_os = "linux"))]
```

Comment:
Instead of removing the test case on Linux, would it be possible to skip it with an appropriate message?

Reply:
We could do it with "guards" (#113).